include/envoy/request_id_extension/request_id_extension.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+#include "envoy/http/header_map.h"
+
+namespace Envoy {
+namespace RequestIDExtension {
+
+enum class TraceStatus { NoTrace, Sampled, Client, Forced };
+
+/**
+ * Abstract request id utilities for getting/setting the request IDs and tracing status of requests
+ */
+class Utilities {
```

Comment:
If you end up moving this into Http, I would call this `ReuestIdExtension` as the interface. If we keep an independent thing, I would probably call the namespace `RequestID` and call this `Extension` or similar?